components/query/query-vector.service.js

```diff
@@ -82,6 +81,38 @@ export default function (
     }
     me.createFeatureAttributeList();
   });
+  $rootScope.$on(
+    'vectorQuery.featureSelected',
+    HsUtilsService.debounce(
+      (event, feature) => {
+        const layer = feature.getLayer(HsMapService.map);
+        if (angular.isDefined(layer) && layer.get('onFeatureSelected')) {
+          const originalFeature = me
+            .getSingleFeature(feature)
+            .map((feature) => feature)
+            .pop();
+          if (originalFeature) {
+            layer.get('onFeatureSelected')(originalFeature);
+          }
+        }
+      },
+      150,
+      false,
+      me
+    )
+  );
+  me.getSingleFeature = function (feature) {
+    let original;
+    if (angular.isDefined(feature)) {
+      original = feature;
+      original.get('features')
+        ? original.get('features').length == 1
+          ? (original = original.get('features'))
+          : (original = null)
+        : null;
+      return original;
+    }
+  };
```

Comment:
This function returns undefined when `feature` is not defined. I have no idea when it can happen, but it might be the source of the bug I observe.
Also, when `feature === undefined`, then the function returns `void`, which is technically `undefined`. So there is no need to check for `if (angular.isDefined(feature))` before `return`. The `return` statement can go out of this block along with       `original = feature;` assignment.

Investigating a bit further I don't think I understand the logic of this function at all... But the returned `null `seems to be causing the error.

Reply:
This code has already been reconstructed btw. Also  the logic is - if feature  is defined, original = feature; this is done so that if the layer is not clustered, code doesn't break. Next I check if feature.get('features'), that means, that the layer is clustered. Than I need to know, if the selected feature has only one clustered source feature, if so, I set original = feature and return it.
![code](https://user-images.githubusercontent.com/46564459/98520394-a4225500-227a-11eb-95d9-52c2e8933898.png)

Reply:
The functionality remains a bit obscured to me, but I made the code work now, so... This is resolved :-)